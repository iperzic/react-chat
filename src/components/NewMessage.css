.NewMessage {
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0;
    height: 60px;
    background: #3798d4;
}

.NewMessage__container{
    height: 100%;
    max-width: 640px;
    margin: 0 auto;
    padding: 8px;
    display: flex;
    box-sizing: border-box;
}

.NewMessage__input, .NewMessage__button {
    box-sizing: border-box;
    border-radius: 5px;
}

.NewMessage__input{
    margin-right: 8px;
    font-size: 20px;
    border: 1px solid #2e77a4;
    padding-left: 8px;
    flex: 8;
}

.NewMessage__button {
    flex: 1;
    background: #ff876d;
    color: white;
    font-size: 15px;
}

.NewMessage__button:disabled {
    background: darkgray;
    color: lightgrey;
}
