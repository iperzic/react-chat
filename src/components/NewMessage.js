import React from 'react';
import PropTypes from 'prop-types';

import './NewMessage.css';

class NewMessage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      message: '',
    };

    this.handleSendMessage = this.handleSendMessage.bind(this);
    this.handleMessageTyping = this.handleMessageTyping.bind(this);
  }

  handleMessageTyping(event) {
    const message = event.currentTarget.value;

    this.setState({ message });
  }

  handleSendMessage(event) {
    event.preventDefault();
    this.props.onMessageSend(this.state.message);
    this.setState({ message: '' });
  }

  render() {
    return (
      <div className="NewMessage">
        <form className="NewMessage__container" onSubmit={this.handleSendMessage}>
          <input
            className="NewMessage__input"
            required
            type="text"
            placeholder=" Message"
            onChange={this.handleMessageTyping}
            value={this.state.message}
            maxLength="256"
          />
          <button
            className="NewMessage__button"
            type="submit"
          >
            Send
          </button>
        </form>
      </div>);
  }
}

NewMessage.propTypes = {
  onMessageSend: PropTypes.func.isRequired,
};

export default NewMessage;
