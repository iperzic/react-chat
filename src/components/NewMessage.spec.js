import React from 'react';
import { shallow } from 'enzyme';

import NewMessage from './NewMessage';

describe('NewMessage', () => {
  let component;
  let messageSendSpy;

  beforeEach(() => {
    messageSendSpy = jest.fn();
    component = shallow(<NewMessage onMessageSend={messageSendSpy} />);
  });

  describe('when there is no text in the input', () => {
    it('should show an input for message and a disabled send button', () => {
      expect(component.getElement()).toMatchSnapshot();
    });

    it('should not call the onMessageSend prop', () => {
      const sendButton = component.find('button');

      sendButton.simulate('click');

      expect(messageSendSpy).not.toHaveBeenCalled();
    });
  });

  describe('when there is text in the input', () => {
    let message;
    beforeEach(() => {
      message = 'Whats up doc?';
      component.setState({ message });
    });

    it('should show an input for message and a send button', () => {
      expect(component.getElement()).toMatchSnapshot();
    });

    it('should call the onMessageSend prop with the message', () => {
      const messageInput = component.find('.NewMessage__input');
      const form = component.find('.NewMessage__container');

      messageInput.simulate('change', {
        currentTarget: {
          value: message,
        },
      });
      form.simulate('submit', {
        preventDefault: () => {
        },
      });

      expect(messageSendSpy).toHaveBeenCalledWith(message);
    });
  });
});
