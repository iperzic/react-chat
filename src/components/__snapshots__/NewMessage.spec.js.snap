// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMessage when there is no text in the input should show an input for message and a disabled send button 1`] = `
<div
  className="NewMessage"
>
  <form
    className="NewMessage__container"
    onSubmit={[Function]}
  >
    <input
      className="NewMessage__input"
      maxLength="256"
      onChange={[Function]}
      placeholder=" Message"
      required={true}
      type="text"
      value=""
    />
    <button
      className="NewMessage__button"
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;

exports[`NewMessage when there is text in the input should show an input for message and a send button 1`] = `
<div
  className="NewMessage"
>
  <form
    className="NewMessage__container"
    onSubmit={[Function]}
  >
    <input
      className="NewMessage__input"
      maxLength="256"
      onChange={[Function]}
      placeholder=" Message"
      required={true}
      type="text"
      value="Whats up doc?"
    />
    <button
      className="NewMessage__button"
      type="submit"
    >
      Send
    </button>
  </form>
</div>
`;
